js/ui/appActivation.js

```diff
@@ -84,26 +87,27 @@ const AppActivationContext = new Lang.Class({
         Main.overview.hide();
     },
 
-    showSplash: function() {
-        // Don't show splash screen unless the launch maximized key is true
+    showSplash: function(launchReason=LAUNCH_REASON_UNINTERESTING) {
         let info = this._app.get_app_info();
-        if (!(info && info.has_key(LAUNCH_MAXIMIZED_DESKTOP_KEY) &&
-              info.get_boolean(LAUNCH_MAXIMIZED_DESKTOP_KEY))) {
-            return;
-        }
-
-        // Don't show splash screen if default maximize is disabled
-        if (global.settings.get_boolean(WindowManager.NO_DEFAULT_MAXIMIZE_KEY)) {
-            return;
-        }
 
-        // Don't show splash screen if this is a link and the browser is
-        // running. We can't rely on any signal being emitted in that
-        // case, as links open in browser tabs.
-        if (this._app.get_id().indexOf('eos-link-') != -1 &&
-            Util.getBrowserApp().state != Shell.AppState.STOPPED) {
+        let dontShowSplashConditions = (
+            // Don't show splash screen unless the launch maximized key is true
+            !(info && info.has_key(LAUNCH_MAXIMIZED_DESKTOP_KEY) &&
+              info.get_boolean(LAUNCH_MAXIMIZED_DESKTOP_KEY)) ||
+            // Don't show splash screen if default maximize is disabled
+            global.settings.get_boolean(WindowManager.NO_DEFAULT_MAXIMIZE_KEY) ||
+            // Don't show splash screen if this is a link and the browser is
+            // running. We can't rely on any signal being emitted in that
+            // case, as links open in browser tabs.
+            (this._app.get_id().indexOf('eos-link-') != -1 &&
+             Util.getBrowserApp().state != Shell.AppState.STOPPED)
+        );
```

Comment:
Please keep the existing early-return style in the code, and split the checks into a `_shouldShowSplash()` function that returns `true` immediately when `launchReason == LAUNCH_REASON_CODING_BUILDER`